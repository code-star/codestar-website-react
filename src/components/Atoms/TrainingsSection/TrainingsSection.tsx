import React, { SFC } from 'react';
import compose from 'recompose/compose';
import Section from '../../../Section/Section';
import { translate, TranslationFunction } from 'react-i18next';
import { Button, Typography } from '@material-ui/core';
import ResponsiveImage from '../../../ResponsiveImage/ResponsiveImage';
import Container from '../../../Container/Container';
import InlineLogo from '../../../InlineLogo/InlineLogo';

// TODO unit test, storybook

// TODO fix width

interface IPropsInner {
  t: TranslationFunction;
}

interface IPropsOuter {
  scrollname: any;
}

export const TrainingsSection: SFC<IPropsInner & IPropsOuter> = ({
  t,
  scrollname,
}) => (
  <Section scrollname={scrollname}>
    <ResponsiveImage
      path="/images/jobs/codestar_4.png"
      asBackgroundImage
      alt="Person receiving training at Codestar"
      effect="/e_grayscale/co_rgb:0057ae,e_colorize:40"
    />
    <Container fullHeight center>
      <div className="row">
        <div className="col-12 col-lg-6">
          <div className="mt-4">
            <Typography
              variant="display3"
              className={`d-inline text-white p-2 bg-dark`}
            >
              {t('TRAININGS')}
            </Typography>
          </div>
          <div className="my-3">
            <Typography
              variant="headline"
              className={`d-inline text-white p-2`}
            >
              RxJS, Scala, TypeScript {t('TRAININGS')}
            </Typography>
          </div>

          <div className="my-2">
            <Button variant="raised">
              {t('TRAININGS')} (contact us for bookings) // TODO
            </Button>
          </div>
          {/*{props.readMore && (
            <div className="my-2">
              <Button
                variant="raised"
                onClick={props.callback}
                className={classes.button}
              >
                {t('CASES_READ_MORE_BUTTON')}
              </Button>
            </div>
          )}*/}
        </div>
        <div className="col-12 col-md-6">
          <div className="bg-dark p-3">
            <Typography variant="body1" className="d-inline text-white">
              <InlineLogo dark>
                {t('TRAININGS')} Lorem ipsum dolor sit amet, consectetur
                adipisicing elit. Aliquid aperiam assumenda commodi deleniti
                doloremque earum eligendi error eum hic incidunt, ipsa itaque
                nihil quas quia quo, repellat tempora veniam voluptate.
              </InlineLogo>
            </Typography>
          </div>
        </div>
      </div>
    </Container>
  </Section>
);

export default compose<IPropsInner, IPropsOuter>(
  translate(['events'], { wait: true })
)(TrainingsSection);
