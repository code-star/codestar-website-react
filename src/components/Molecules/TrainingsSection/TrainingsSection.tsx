import React, { SFC } from 'react';
import compose from 'recompose/compose';
import Section from '../Section/Section';
import { translate, TranslationFunction } from 'react-i18next';
import { Tooltip, Typography, Grow, Avatar } from '@material-ui/core';
import { Link } from 'react-router-dom';
import { CustomButton } from '../../Atoms/CustomButton/CustomButton';
import ResponsiveImage, {
  getResponsiveImageUrl,
} from '../../../ResponsiveImage/ResponsiveImage';
import Container from '../../../Container/Container';
import styles from './TrainingsSection.module.scss';

interface IPropsInner {
  t: TranslationFunction;
}

interface IPropsOuter {
  scrollname: any;
}

const getTechIcon: any = (name: string, url: string) => (
  <div className="col d-flex justify-content-center">
    <Tooltip title={name}>
      <Grow in>
        <Avatar alt={name} src={getResponsiveImageUrl(url, 80)} />
      </Grow>
    </Tooltip>
  </div>
);

export const TrainingsSection: SFC<IPropsInner & IPropsOuter> = ({
  t,
  scrollname,
}) => (
  <Section scrollname={scrollname} className={styles.section}>
    <ResponsiveImage
      path="/images/jobs/codestar_4.png"
      asBackgroundImage
      alt="Person receiving training at Codestar"
      effect="/e_grayscale/co_rgb:0057ae,e_colorize:40"
    />
    <Container fullHeight center>
      <div className="row">
        <div className="col-12 col-lg-6">
          <div className="mt-4">
            <Typography
              variant="h2"
              className={`d-inline text-white p-2 bg-dark`}
            >
              {t('TRAININGS')}
            </Typography>
          </div>
          <div className="my-3">
            <Typography
              variant="h5"
              className={`d-inline text-white p-2`}
            >
              RxJS, Scala, TypeScript {t('TRAININGS')}
            </Typography>
          </div>

          <div className="my-2">
            <CustomButton variant="contained" to="/contact" component={Link}>
              {t('BOOK_A_TRAINING')}
            </CustomButton>
          </div>
        </div>
        <div className="col-12 col-md-6">
          <div className="bg-dark p-3">
            <Typography className="d-inline text-white">
              {t('TRAININGS_TEXT')}
            </Typography>
            <div className="row m-1 mt-2">
              {getTechIcon('Scala', '/images/technologies/scala.svg')}
              {getTechIcon('Kotlin', '/images/technologies/kotlin.svg')}
              {getTechIcon('RxJS', '/images/technologies/rxjs.svg')}
              {getTechIcon('TypeScript', '/images/technologies/typescript.svg')}
            </div>
          </div>
        </div>
      </div>
    </Container>
  </Section>
);

export default compose<IPropsInner & IPropsOuter, IPropsOuter>(
  translate(['events'], { wait: true })
)(TrainingsSection);
