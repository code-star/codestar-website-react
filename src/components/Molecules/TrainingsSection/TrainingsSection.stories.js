import React from 'react'
import { storiesOf } from '@storybook/react'
import { MemoryRouter } from 'react-router-dom'
import TrainingsSection from './TrainingsSection'

const stories = storiesOf('components/Molecules/TrainingsSection', module)

stories.add('TrainingsSection', () => (
  <MemoryRouter>
    <TrainingsSection />
  </MemoryRouter>
))
