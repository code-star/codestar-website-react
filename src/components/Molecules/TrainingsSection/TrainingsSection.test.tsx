import React from 'react';
import { shallow } from 'enzyme';
import TrainingsSection from './TrainingsSection';
import { MemoryRouter } from 'react-router';

jest.mock('react-i18next', () => ({
  translate: () => (Component: any) => {
    Component.defaultProps = Component.defaultProps || {};
    Component.defaultProps.t = () => '';
    return Component;
  },
}));

jest.mock('@material-ui/core/Grow', () => (Component: any) => {
  const React = require('react');
  return <div>{Component.children}</div>;
});

const globalAny: any = global;

const renderShallow = () => {
  return shallow(
    <MemoryRouter>
      <TrainingsSection scrollname="trainings" />
    </MemoryRouter>
  );
};

describe('<TrainingsSection />', () => {
  describe('Snaphot', () => {
    test('must match the static training section', () => {
      expect(globalAny.renderToJSON(renderShallow())).toMatchSnapshot();
    });
  });
});
