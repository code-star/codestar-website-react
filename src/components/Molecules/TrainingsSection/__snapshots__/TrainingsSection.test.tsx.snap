// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrainingsSection /> Snaphot must match the static training section 1`] = `
<section
  className="
			section
			WithTheme(WithWidth(Section))-section"
  name="trainings"
>
  <div
    className="WithTheme(WithWidth(Section))-element"
    name="trainings"
  />
  <img
    alt="Person receiving training at Codestar"
    className="asBackgroundImage"
    sizes="100w"
    src="/images/jobs/codestar_4.png"
    srcSet="https://res.cloudinary.com/codestar/image/upload/f_auto/w_375//e_grayscale/co_rgb:0057ae,e_colorize:40/v1542784068/codestar.nl/images/jobs/codestar_4.png 375w, https://res.cloudinary.com/codestar/image/upload/f_auto/w_800//e_grayscale/co_rgb:0057ae,e_colorize:40/v1542784068/codestar.nl/images/jobs/codestar_4.png 800w, https://res.cloudinary.com/codestar/image/upload/f_auto/w_1280//e_grayscale/co_rgb:0057ae,e_colorize:40/v1542784068/codestar.nl/images/jobs/codestar_4.png 1280w, https://res.cloudinary.com/codestar/image/upload/f_auto/w_1536//e_grayscale/co_rgb:0057ae,e_colorize:40/v1542784068/codestar.nl/images/jobs/codestar_4.png 1536w, https://res.cloudinary.com/codestar/image/upload/f_auto/w_1920//e_grayscale/co_rgb:0057ae,e_colorize:40/v1542784068/codestar.nl/images/jobs/codestar_4.png 1920w"
    title=""
  />
  <div
    className="
			container
			
			
			WithTheme(WithWidth(Container))-fullHeight
			
			
			WithTheme(WithWidth(Container))-containerCenter
		"
  >
    <div
      className="row"
    >
      <div
        className="col-12 col-lg-6"
      >
        <div
          className="mt-4"
        >
          <h1
            className="MuiTypography-root MuiTypography-display3 d-inline text-white p-2 bg-dark"
          >
            
          </h1>
        </div>
        <div
          className="my-3"
        >
          <h1
            className="MuiTypography-root MuiTypography-headline d-inline text-white p-2"
          >
            RxJS, Scala, TypeScript 
            
          </h1>
        </div>
        <div
          className="my-2"
        >
          <a
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-raised"
            href="/contact"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
          >
            <span
              className="MuiButton-label"
            >
              
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
      <div
        className="col-12 col-md-6"
      >
        <div
          className="bg-dark p-3"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 d-inline text-white"
          >
            
          </p>
          <div
            className="row m-1 mt-2"
          >
            <div
              className="col d-flex justify-content-center"
            >
              <div>
                <div
                  className="MuiAvatar-root"
                >
                  <img
                    alt="Scala"
                    className="MuiAvatar-img"
                    src="https://res.cloudinary.com/codestar/image/upload/f_auto/w_80/v1542784068/codestar.nl/images/technologies/scala.svg"
                  />
                </div>
              </div>
            </div>
            <div
              className="col d-flex justify-content-center"
            >
              <div>
                <div
                  className="MuiAvatar-root"
                >
                  <img
                    alt="Kotlin"
                    className="MuiAvatar-img"
                    src="https://res.cloudinary.com/codestar/image/upload/f_auto/w_80/v1542784068/codestar.nl/images/technologies/kotlin.svg"
                  />
                </div>
              </div>
            </div>
            <div
              className="col d-flex justify-content-center"
            >
              <div>
                <div
                  className="MuiAvatar-root"
                >
                  <img
                    alt="RxJS"
                    className="MuiAvatar-img"
                    src="https://res.cloudinary.com/codestar/image/upload/f_auto/w_80/v1542784068/codestar.nl/images/technologies/rxjs.svg"
                  />
                </div>
              </div>
            </div>
            <div
              className="col d-flex justify-content-center"
            >
              <div>
                <div
                  className="MuiAvatar-root"
                >
                  <img
                    alt="TypeScript"
                    className="MuiAvatar-img"
                    src="https://res.cloudinary.com/codestar/image/upload/f_auto/w_80/v1542784068/codestar.nl/images/technologies/typescript.svg"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
